docs: explain https support (#4500)
jina-ai/jina: docs/fundamentals/flow/client.md

The most convenient way to work with the `Flow` API is the Python Client.
It enables you to send `Documents` to a running `Flow` in a number of different ways, as shown below.

## HTTP, gRPC, and WebSocket

Jina Flows and Clients support three different networking protocols: HTTP, gRPC, and WebSocket.
For each of them, you first connect your Client to the Flow, before you can send requests to it.

### Connect Client to a Flow

If there is not already a Flow running in the background or on the network, you can start one:

```python
from jina import Flow

PORT = 1234
PROTOCOL = 'grpc'  # one of 'grpc', 'http', 'websocket'

with Flow(port=PORT, protocol=PROTOCOL) as f:
    f.block()
```

To connect to the `Flow`, the Client has to specify the followings parameters.
All af these have to match the Flow and how it was set up:
* the `protocol` it needs to use to communicate with the Flow

> * the `host` and the `port`on which the Flow is exposed

Reviewer: ```suggestion
* the `host` and the `port` as exposed by the Flow
```

````{admonition} Caution
:class: caution
You can't define these parameters both by keyword argument and by host scheme - you can't have two sources of truth.
Example: the following code will raise an exception:
```python
Client(host='https://my.awesome.flow:1234', port=4321)
```
````

````{admonition} Hint
:class: hint
The arguments above have usefule defaults: `protocol='grpc'` and `host='0.0.0.0'`.
This is particularly useful when debugging or accessing a Flow on your local machine.

> To connect to a Flow `f` it is therefore often enought to do the following:

Reviewer: ```suggestion
To connect to a Flow `f` it is therefore often enough to do the following:
```

```{code-block} python
c = Client(port=f.port)
```
````

### Send requests to the Flow

After a Client has connected to a Flow, it can send requests to the Flow using its `.post()` method.
This expects as inputs the {ref}`Executor endpoint <exec-endpoint>` that you want to target, as well as a Document or Iterable of Documents:

```python
from docarray import Document, DocumentArray

d1 = Document(content='hello')
d2 = Document(content='world')

def doc_gen():
    for j in range(10):
        yield Document(content=f'hello {j}')

client = Client(port=PORT)

client.post('/endpoint', d1)  # Single Document

client.post('/endpoint', [d1, d2])  # List of Documents

client.post('/endpoint', doc_gen)  # Document generator

client.post('/endpoint', DocumentArray([d1, d2]))  # DocumentArray

client.post('/endpoint')  # Empty
```

```{admonition} Caution
:class: caution
`Flow` also provides a `.post()` method that follows the same interface as `client.post()`.

> However, once your solution is deployed in the cloud, the Flow interface is not present anymore.

Reviewer: ```suggestion
However, once your solution is deployed remotely, the Flow interface is not present anymore.
```
